weather.go

```diff
@@ -1,5 +1,127 @@
 package weather
 
-func Get(location, token string) ([]byte, error) {
-	return nil, nil
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+)
+
+type Weather struct {
+	Coord struct {
+		Lon float64 `json:"lon"`
+		Lat float64 `json:"lat"`
+	} `json:"coord"`
+	Weather []struct {
+		ID          int    `json:"id"`
+		Main        string `json:"main"`
+		Description string `json:"description"`
+		Icon        string `json:"icon"`
+	} `json:"weather"`
+	Base string `json:"base"`
+	Main struct {
+		Temp      float64 `json:"temp"`
+		FeelsLike float64 `json:"feels_like"`
+		TempMin   float64 `json:"temp_min"`
+		TempMax   float64 `json:"temp_max"`
+		Pressure  int     `json:"pressure"`
+		Humidity  int     `json:"humidity"`
+	} `json:"main"`
+	Visibility int `json:"visibility"`
+	Wind       struct {
+		Speed float64 `json:"speed"`
+		Deg   int     `json:"deg"`
+	} `json:"wind"`
+	Clouds struct {
+		All int `json:"all"`
+	} `json:"clouds"`
+	Dt  int `json:"dt"`
+	Sys struct {
+		Type    int    `json:"type"`
+		ID      int    `json:"id"`
+		Country string `json:"country"`
+		Sunrise int    `json:"sunrise"`
+		Sunset  int    `json:"sunset"`
+	} `json:"sys"`
+	Timezone int    `json:"timezone"`
+	ID       int    `json:"id"`
+	Name     string `json:"name"`
+	Cod      int    `json:"cod"`
+}
+
+type CityUnknown struct {
+	Cod     string `json:"cod"`
+	Message string `json:"message"`
+}
+
+type CliResponse struct {
+	OneWord string  `json:"oneword"`
+	Celcius float64 `json:"celcius"`
+}
+
+func CallUrl(token, location string) http.Response {
+
+	var cu CityUnknown
+
+	resp, err := http.Get(fmt.Sprintf("https://api.openweathermap.org/data/2.5/weather?q=%s&appid=%s", location, token))
```

Comment:
If this `api.openweathermap.org` domain is hard-wired into the `CallUrl` function, it makes a bit difficult to test using our local test server, doesn't it? Maybe we should pass in the domain part as a parameter instead. Then the test could call it using `ts.URL` instead of the real API's address.

Reply:
not sure if I have done what you suggested here...

Reply:
✅